resources/ResizeService.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace ImageFocus;
+
+/**
+ * The class responsible for resizing the images trough default WordPress functionality
+ *
+ * Class ResizeService
+ * @package ImageFocus
+ */
+class ResizeService
+{
+    public function __construct()
+    {
+        $this->addHooks();
+    }
+
+    /**
+     * Make sure all hooks are being executed.
+     */
+    private function addHooks()
+    {
+        add_filter('intermediate_image_sizes_advanced', [$this, 'resizeAttachments'], 10, 2);
+    }
+
+    /**
+     * Catch the intermediate_image_sizes_advanced filter.
+     * And make sure all the resizing goes trough the ImageFocus crop service.
+     *
+     * @param $sizes
+     * @param $metadata
+     */
+    public function resizeAttachments($sizes, $metadata)
+    {
+        // The filter doesn't pass trough the attachment's ID. So we need to get it by url
+        $attachmentId = $this->getAttachtmentIdByUrl($metadata['file']);
+
+        // Get the focus point
+        $focusPoint = get_post_meta($attachmentId, 'focus_point', true);
+
+        // Crop the attachment trough the crop service
+        $crop = new CropService();
+        $crop->crop($attachmentId, $focusPoint);
+
+        foreach ($sizes as $size => $image) {
+            if ($image['crop'] === 1) {
+                unset($sizes[$size]);
+            }
+        }
+    }
+
+    /**
+     * Get the attachment's ID by URL
+     *
+     * @param $file
+     * @return mixed
+     */
+    private function getAttachtmentIdByUrl($file)
+    {
+        global $wpdb;
+        $attachment = $wpdb->get_col(
+            $wpdb->prepare("SELECT ID FROM $wpdb->posts WHERE guid='%s';", wp_upload_dir()['baseurl'] . '/' . $file)
+        );
+
+        return $attachment[0];
```

Comment:
it is impossible for this method not to have a [0] element. But I will improve it.